src/matrices.jl

```diff
@@ -39,10 +39,18 @@ function Base.similar(mat::pm_Matrix, ::Type{S}, dims::Dims{2}) where S<:Union{p
     return pm_Matrix{convert_to_pm_type(S)}(dims...)
 end
 
-function Base.similar(mat::pm_Matrix, ::Type{S}, dims::Dims{1}) where S<:Union{pm_Integer, pm_Rational, Float64}
+function Base.similar(mat::pm_Matrix, ::Type{S}, dims::Dims{2}) where S
+    return Matrix{S}(undef, dims...)
+end
+
+function Base.similar(mat::pm_Matrix, ::Type{S}, dims::Dims{1}) where S<:Union{pm_Integer, pm_Rational}
```

Comment:
Can you remove the `Float64` type in the above definition for `Dims{2}` to keep the symmetry in the definitions?

Reply:
If `M` is a `pm_Matrix{pm_Rational}` slicing `M[1, :]` results in `pm_Vector{pm_Rational}`, and `M[[1], :]` returns a `pm_Matrix{pm_Rational}` (as expected). However if `M` is a `pm_Matrix{Float64}` the latter works, but the former fails, since we don't wrap `pm_Vector{Float64}`. This makes `M[1, :]` return `Vector{Float64}` which is the best we can afford (at least until someone wraps `pm_Vector{Float64}`)